### PR TITLE
Make `⌊` account for decimals and negatives

### PR DESCRIPTION
**Pull request by Lyxal**
Goes towards closing #1164

### CODE REVIEW

File: vyxal/elements.py [hunk 1]
```diff
@@ -5658,6 +5658,20 @@ def vy_floor(lhs, ctx):
     (num) -> floor(a)
     (str) -> integer part of a
     """
+
+    def vy_floor_str_helper(item):
```
Comment (cgccuser) [inline vyxal/elements.py:5662]: Wanna move this to helpers.py?

File: vyxal/elements.py [hunk 2]
```diff
@@ -5658,6 +5658,20 @@ def vy_floor(lhs, ctx):
...
+        temp = ""
+        for char in item:
+            if char == "-" and temp == "":
+                temp += char
+            elif char.isdigit():
+                temp += char
+            elif char == "." and temp.count(".") == 0:
```
Comment (cgccuser) [inline vyxal/elements.py:5669]:
```suggestion
            elif char == "." and "." not in temp:
```

File: vyxal/elements.py [hunk 3]
```diff
@@ -5658,6 +5658,20 @@ def vy_floor(lhs, ctx):
...
+                temp += char
+            else:
+                continue
```
Comment (cgccuser) [inline vyxal/elements.py:5672]:
```suggestion
```
No need to continue if you're already at the end.

File: vyxal/elements.py (hunk 2 above)
Comment (cgccuser) [reply vyxal/elements.py:5669]: Might want to make sure you don't get a string like `-.` by adding a zero.

**Review by cgccuser — CHANGES_REQUESTED** [review]

**Review by cgccuser — APPROVED** [review]

**Review by Steffan153 — APPROVED** [review]

**Pull request merged**